x-pack/legacy/plugins/lens/server/usage/task.ts

```diff
@@ -0,0 +1,235 @@
+  const { kbnServer } = (server.plugins.xpack_main as XPackMainPlugin & {
+    status: { plugin: { kbnServer: KbnServer } };
+  }).status.plugin;
+
+  kbnServer.afterPluginsInit(() => {
+    // The code block below can't await directly within "afterPluginsInit"
+    // callback due to circular dependency The server isn't "ready" until
+    // this code block finishes. Migrations wait for server to be ready before
+    // executing. Saved objects repository waits for migrations to finish before
+    // finishing the request. To avoid this, we'll await within a separate
+    // function block.
+    (async () => {
+      try {
+        await taskManager!.schedule({
+          id: TASK_ID,
+          taskType: TELEMETRY_TASK_TYPE,
+          state: { byDate: {}, suggestionsByDate: {}, saved: {}, runs: 0 },
+          params: {},
+        });
+      } catch (e) {
+        server.log(['warning', 'telemetry'], `Error scheduling task, received ${e.message}`);
+      }
+    })();
+  });
+}
+
+export async function getDailyEvents(
+  kibanaIndex: string,
+  callCluster: ClusterSearchType & ClusterDeleteType
+): Promise<{
+  byDate: Record<string, Record<string, number>>;
+  suggestionsByDate: Record<string, Record<string, number>>;
+}> {
+  const aggs = {
+    daily: {
+      date_histogram: {
+        field: 'lens-ui-telemetry.date',
+        calendar_interval: '1d',
+        min_doc_count: 1,
+      },
+      aggs: {
+        groups: {
+          filters: {
+            filters: {
+              suggestionEvents: {
+                bool: {
+                  filter: {
+                    term: { 'lens-ui-telemetry.type': 'suggestion' },
+                  },
+                },
+              },
+              regularEvents: {
+                bool: {
+                  must_not: {
+                    term: { 'lens-ui-telemetry.type': 'suggestion' },
+                  },
+                },
+              },
+            },
+          },
+          aggs: {
+            names: {
+              terms: { field: 'lens-ui-telemetry.name', size: 100 },
+              aggs: {
+                sums: { sum: { field: 'lens-ui-telemetry.count' } },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  const metrics: AggregationSearchResponseWithTotalHitsAsInt<
+    unknown,
+    {
+      body: { aggs: typeof aggs };
+    }
+  > = await callCluster('search', {
+    index: kibanaIndex,
+    rest_total_hits_as_int: true,
+    body: {
+      query: {
+        bool: {
+          filter: [
+            { term: { type: 'lens-ui-telemetry' } },
+            { range: { 'lens-ui-telemetry.date': { gte: 'now-90d/d' } } },
+          ],
+        },
+      },
+      aggs,
+    },
+    size: 0,
+  });
+
+  const byDateByType: Record<string, Record<string, number>> = {};
+  const suggestionsByDate: Record<string, Record<string, number>> = {};
+
+  metrics.aggregations!.daily.buckets.forEach(daily => {
```

Comment:
Are we sure this can never be undefined? Same for the deeper properties.

Reply:
As long as the `.kibana` index exists, the `aggregations` key will be returned. And if not, this entire task will fail but that's also handled